### PR TITLE
Update documentation and examples

### PR DESCRIPTION
**Pull request by lmweber**
Ok here is an update for the documentation and examples (starting from `spatialCoordsData` branch):

- update documentation in Roxygen blocks
- update examples

For the examples, I have tried to keep it simple by re-using the one from `read10xVisium()` where possible, and loading it with `example(read10xVisium)`.

I also added the new mock data example of molecule-based data that @HelenaLC on Slack, which demonstrates the `molecules()` functionality.

I haven't done anything in the vignettes, and also didn't update DESCRIPTION to avoid merge conflicts.

Will run the local checks again now, and then do some further checking later today.

### CODE REVIEW

**Review by drighelli — APPROVED** [review]

**Review by HelenaLC — COMMENTED** [review]
in `SpatialExperiment-assays` it says "molecules()<- get or set a BumpyMatrix of raw count data, e.g., number of reads or transcripts" - this is confusing to me. Aren't the number of x still in `counts`, and the `BumpyMatrix` in `molecules_ij` contains locations, i.e. xy-coordinates of molecules from feature i assigned to sample j?
Maybe rephrase to "get or set a BumpyMatrix of spatial molecule positions, i.e. xy-coordinates of molecules from a given feature assigned to a given sample" or similar?

**Review by HelenaLC — COMMENTED** [review]
in `spatialExperiment-colData` is says "expects a DF with a sample_id column..." - this is not the case. The replacement method allows sample IDs to be missing, in which case the old sample IDs will be retained. Meanwhile, if sample IDs are provided, certain validity checks kick in, which I think should be explained in more detail in the documentation & examples. Specifically, two things would be worth mentioning:
- replacement by NULL will always retain sample IDs
- replacing sample IDs only works if the number of unique values is the same, and there's a one-to-one mapping between old and new IDs
Maybe even worth adding an invalid replacement example in the dox with tryCatch?

**Review by HelenaLC — COMMENTED** [review]
`SpatialExperiment-methods` still mentions deprecated arguments: `cd_bind` and `as_df`. 
- `cd_bind` is not `colData` for simplicity, along with `spatialCoords` to include spatial coordinates
- `as_df` is dropped for consistency; we always return a `DataFrame`

**Review by HelenaLC — COMMENTED** [review]
Veeery minor, but the return value of `show` is actually NULL... Hence many packages write "none" as @return. You can see then when doing e.g.
```
> foo <- show(1)
[1] 1
> foo
NULL
```

**Comment by lmweber** [comment]
Thanks for these comments. I have pushed another commit above with additional updates to address these. This also passes local checks.

**Review by drighelli — COMMENTED** [review]
why the documentation is changed and the documentation files are not?

Have you run the document() command?

**Comment by HelenaLC** [comment]
They are, no? I can see the .Rd man files listed under "files changed"

**Comment by drighelli** [comment]
> They are, no? I can see the .Rd man files listed under "files changed"

I noticed this because I made a first review before your comments and I approved them.

After the latest review github says that they are not changed, that's why I was asking...

Anyway, we can start to merge for me! :)

Thanks

**Comment by lmweber** [comment]
Yes, documentation files are all there. I ran `devtools::document()`, `devtools::install()`, `devtools::check()`, and `BiocCheck::BiocCheck()` in my local installation.

We will also need to look at the build reports afterwards though, in case there are unexpected differences in the local installation vs. GHA vs. Bioc builders.
